oda_api/api.py

```diff
@@ -903,18 +903,8 @@ def get_product(self,
             raise RuntimeError(
                 "not failed, not, but complete? programming error for client!")
 
-        # <
-
-        data = None
-
-        if not dry_run:
-            d = DataCollection.from_response_json(
-                res_json, instrument, product)
-
-        else:
-            self._decode_res_json(
-                res.json()['products']['instrument_parameters'])
```

Comment:
you may deduce that this did not work since above `res.json()` was a  list

Reply:
indeed it was the point where the crash related to #84 was happening

Reply:
I do not completely understand why we did not see it before. This should have happened with all of the email API codes. Though only when product is ready.

Reply:
Can we make a test to make sure this completes?..

Reply:
Sure. But, should we assume that perhaps, in some cases, `dry_run` is still passed as a parameter to the `oda_api` ?

Reply:
Because in that case, we have to make sure that it won't be included within the output api_code

Reply:
Ok, and add a warning that it does nothing, and the request will be checked when it is sent.

I think dry_run functionality - checking parameters was previously done by dispatcher. Now list of valid parameters is inspected by oda_api.

Reply:
Indeed, in the dispatcher simply lists the parameters:
[https://github.com/oda-hub/dispatcher-app/blob/c0dcd03461fd7e1bb7aa9591814f46bb11ab05b8/cdci_data_analysis/analysis/instrument.py#L227](url)